src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -148,12 +149,13 @@ public function transChoice($id, $number, array $parameters = array(), $domain =
      * @param array $parameters
      * @param string $domain
      * @param string|null $locale
+     * @param string|null $legacy
      *
      * @return mixed|string
      *
      * @throws \Exception
      */
-    private function translateUsingLegacySystem($message, array $parameters, $domain, $locale = null)
+    private function translateUsingLegacySystem($message, array $parameters, $domain, $locale = null, $legacy = null)
```

Comment:
I'm not sure why it would be useful to let the legacy callback fall through. I wouldn't even let it work with `trans()` if removing it wasn't a breaking change.